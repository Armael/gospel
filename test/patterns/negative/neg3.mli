type t7 = char * char

val f7 : t7 -> int
(*@ r = f7 a
    ensures
      match a with
      | '\000', x -> true
      | x, '\000' -> true
*)
