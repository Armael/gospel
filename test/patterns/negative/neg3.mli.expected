
*******************************
********** Parsed file ********
*******************************
type t7 = (char * char)
val f7 : t7 -> int[@@gospel
                    {| r = f7 a
    ensures
      match a with
      | '\000', x -> true
      | x, '\000' -> true
|}]

*******************************
****** GOSPEL translation *****
*******************************
(*@ open Stdlib *)

(*@ open Gospelstdlib *)

type t7 = (char * char)
  

val f7 : t7 -> int
(*@ r = f7 a
    ensures ...
     *)
File "neg3.mli", line 6, characters 6-70:
Error: This pattern-matching is not exhaustive.
       Here is an example of a case that is not matched:
         '\001', '\001'.
